### ISSUE TITLE
Support spaces in unix sockets

### ISSUE THREAD

**Issue by Paxa** [vitaly-t/connection-string#25]
Now it will truncate until space

```js
ConnectionString.parseHost('/Users/pavel/Library/Application Support/Postgres/var-10') => 
{name: "/Users/pavel/Library/Application", type: "socket"}
```

**Comment by vitaly-t** [comment]
Interesting. I haven't come across use of Unix sockets with spaces in them. Do you have a spec reference to indicate that spaces can be used inside socket names?

If so, then it will need to be fixed.

**Comment by Paxa** [comment]
afaik any folder may contain socket file, and socket itself can be named as any other file.
So probably it should support some special chars e.g. `-_,$` and utf8 symbols of different alphabets

**Comment by Paxa** [comment]
I made a workaround for that https://github.com/Paxa/postbird/blob/3ccf8fa72d49675367c817c1525af07673e34186/app/connection.js#L142-L147

**Comment by vitaly-t** [comment]
In your work-around, you are using a port with the socket? Sockets do not support ports, AFAIK.

**Comment by Paxa** [comment]
Postgres socket file named `.s.PGSQL.5432`, and if we set `host = /tmp` then it will try to use `/tmp/.s.PGSQL.5432`.
But if postgres run on different port, then socket filename will be different

**Comment by vitaly-t** [comment]
That's still just a socket name, not a port. It just a coincidence that you are adding port to the socket name, that is not a standard.

**Comment by Paxa** [comment]
Agree, socket name should be a socket file name, not a folder.

But it seems like a standard behaviour for `psql` and `node-postgres` make it same for consistency
https://github.com/brianc/node-postgres/blob/e4578d2c7bca60cfb02ec004daa22c8ff810140e/lib/client.js#L97

I test like this:
1. Run postgres on custom port, e.g. 5435
2. `ls /tmp/.s.PGSQL*` => `/tmp/.s.PGSQL.5435      /tmp/.s.PGSQL.5435.lock`
3. Try with psql
```
# psql
psql: error: could not connect to server: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/tmp/.s.PGSQL.5432"?
```
4. Try with socket name
```
psql -h /tmp/.s.PGSQL.5435        
psql: error: could not connect to server: could not connect to server: Not a directory
	Is the server running locally and accepting
	connections on Unix domain socket "/tmp/.s.PGSQL.5435/.s.PGSQL.5432"?
```
5. It works as
```
psql -p 5435
psql -h /tmp -p 5435
```
6. Make sure we are connected via unix socket not tcp
```
# psql -p 5435
psql (12.0)
Type "help" for help.

me=# SELECT application_name, client_addr, client_port FROM pg_stat_activity where pid = pg_backend_pid();
 application_name | client_addr | client_port 
------------------+-------------+-------------
 psql             |             |          -1
(1 row)
```

**Comment by vitaly-t** [comment]
The problem is, there is no syntax within URL Connection String for defining ports for sockets. Ports for regular hosts are defined with `:` prefix. And in your name you are using some random `.`+port.

There is no standard that can be implemented.

**Comment by Paxa** [comment]
Yep, this is specific for postgres, and probably should remain as workaround. (and outside of this issue scope)
Btw where did you find standard for URL Connection String?

**Comment by vitaly-t** [comment]
> where did you find standard for URL Connection String?

First link you find in the [WiKi](https://github.com/vitaly-t/connection-string/wiki).

**Comment by vitaly-t** [comment]
Fixed in [v3.1.0](https://github.com/vitaly-t/connection-string/releases/tag/3.1.0) - see the release notes.

Closing it now, since the main issue has been resolved.

@Paxa If you you think there's any point in discussing the issue with the port inside a socket name, feel free opening a separate issue for that.

**Comment by vitaly-t** [comment]
@Paxa I implemented this functionality, following your report. Could you, please, follow up, whether the issue was completely resolved on your side?
